Eatery/Views/Eateries/CollegetownMenuHeaderView.swift

```diff
@@ -0,0 +1,337 @@
+            make.height.equalTo(17)
+        }
+        
+        cuisineLabel.snp.makeConstraints { make in
+            make.top.equalTo(statusLabel.snp.bottom).offset(2)
+            make.leading.equalTo(statusLabel)
+            make.height.equalTo(45)
+            make.width.equalTo(260)
+        }
+        
+        locationLabel.snp.makeConstraints { make in
+            make.top.equalTo(cuisineLabel.snp.bottom).offset(6)
+            make.leading.equalTo(cuisineLabel)
+            make.trailing.lessThanOrEqualTo(informationView)
+            make.height.equalTo(17)
+        }
+        
+        ratingView.snp.makeConstraints { make in
+            make.centerY.equalTo(statusLabel)
+            make.trailing.equalTo(informationView).inset(12)
+        }
+        
+        priceLabel.snp.makeConstraints { make in
+            make.top.equalTo(ratingView.snp.bottom).offset(13)
+            make.trailing.equalToSuperview().inset(12.5)
+            make.height.equalTo(19)
+            make.width.equalTo(30.5)
+        }
+        
+        distanceLabel.snp.makeConstraints { make in
+            make.centerY.equalTo(locationLabel)
+            make.trailing.equalToSuperview().inset(12.5)
+            make.height.equalTo(17)
+            make.width.lessThanOrEqualToSuperview()
+        }
+        
+    }
+    
+    func set(eatery: CollegetownEatery, userLocation: CLLocation?) {
+        titleLabel.text = eatery.displayName
+        if let url = eatery.imageUrl {
+            let placeholder = UIImage.image(withColor: UIColor(white: 0.97, alpha: 1.0))
+            backgroundImageView.kf.setImage(with: url, placeholder: placeholder)
+        }
+
+        let eateryStatus = eatery.currentStatus()
+        
+        switch eateryStatus {
+        case .open, .closingSoon:
+            titleLabel.textColor = .white
+            
+        case .closed, .openingSoon:
+            titleLabel.textColor = UIColor.darkGray
+            
+            let closedView = UIView()
+            closedView.backgroundColor = UIColor(white: 1.0, alpha: 0.65)
+            backgroundImageView.addSubview(closedView)
+            closedView.snp.makeConstraints { make in
+                make.edges.equalToSuperview()
+            }
+        }
+        
+        paymentView.paymentMethods = eatery.paymentMethods
+
+        let presentation = eatery.currentPresentation()
+        statusLabel.text = presentation.statusText
+        statusLabel.textColor = presentation.statusColor
+        hourLabel.text = presentation.nextEventText
+        
+        cuisineLabel.text = eatery.categories.joined(separator: ", ")
+        
+        locationLabel.text = eatery.address
+        
+        let star1 = ratingView.ratingImageView[0]
+        let star2 = ratingView.ratingImageView[1]
+        let star3 = ratingView.ratingImageView[2]
+        let star4 = ratingView.ratingImageView[3]
+        let star5 = ratingView.ratingImageView[4]
+
+        if let rating = eatery.rating {
```

Comment:
Following converting rating away from an enum, this could be simplified by iterating from 0 to rating inclusive and setting these images to selected, while making the remaining images unselected. The half selected image can then simply be handled in a check for whether rating can be represented as an integer or only a double, updating the corresponding image if needed.

Reply:
This algorithm should be changed, but let's leave that for another PR.